test/unittest/extension/transpose_test.cpp

```diff
@@ -0,0 +1,134 @@
+/***************************************************************************
+ *
+ *  @license
+ *  Copyright (C) Codeplay Software Limited
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  For your convenience, a copy of the License has been included in this
+ *  repository.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  SYCL-BLAS: BLAS implementation using SYCL
+ *
+ *  @filename transpose.cpp
+ *
+ **************************************************************************/
+
+#include "blas_test.hpp"
+
+using index_t = int;
+
+enum trans_type : int { Inplace = 0, Outplace = 1 };
```

Comment:
Probably, it is convenient to use a `char` (e.g. `i` and `o`) rather than an `enum`. This allows to print out a more understandable test-name, `tr_type_{0,1}` vs `tr_type_{i,o}`.

Moreover, in this way, you can also get rid of `dump_arg`.